spec/requests/posts_spec.rb

```diff
@@ -1,17 +1,22 @@
 require 'rails_helper'
 
 RSpec.describe 'Posts', type: :request do
-  describe 'GET /index' do
-    it 'returns http success' do
-      get '/posts/index'
-      expect(response).to have_http_status(:success)
+  describe 'GET /users/:user_id/posts' do
+    let(:user) { User.create(name: 'Test User') }
+
+    it 'returns a successful response' do
+      get "/users/#{user.id}/posts"
+      expect(response).to have_http_status(200)
+    end
+
+    it 'renders the index template' do
+      get "/users/#{user.id}/posts"
+      expect(response).to render_template(:index)
     end
-  end
 
-  describe 'GET /show' do
-    it 'returns http success' do
-      get '/posts/show'
-      expect(response).to have_http_status(:success)
+    it 'includes correct placeholder text in the response body' do
+      get "/users/#{user.id}/posts"
+      expect(response.body).to include('List of all posts')
     end
```

Comment:
- [ ] @Gardimy Great work adding tests for the posts index page, however, you forgot to add tests for the post-show page, Kindly consider adding tests for this page.